### ISSUE TITLE
Show room name in push notifications from rooms

### ISSUE THREAD

**Issue by sakluger** [Expensify/App#3844]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:

1. I received a push notification for a DM in Expensify.com. The notification shows the name of the person who sent the message in the room along with a snippet from the message they sent.
2. I then received a push notification for a room I am a member of in Expensify.com. Just like the DM push notification, the room notification shows the name of the person who sent the message in the room along with a snippet from the message they sent.

## Expected Result:

Push notifications for rooms should show (1) the room name, (2) the name of the person sending the message, and (3) a snippet from the message they sent.

If the notification doesn't show the room name, then it is impossible to know whether it was sent as a DM or within a room.

Slack does a good job of differentiating between DM and channel notifications:

![image](https://user-images.githubusercontent.com/5997400/124044518-7aad6980-d9c2-11eb-8941-9f649fe2e939.png)

## Actual Result:

Push notifications for rooms only show the name of the person sending the message and a snippet from the message, they don't show the room name. Here is a screenshot of three push notifications I received - the top one is from a room and the other two are from a DM:

![image](https://user-images.githubusercontent.com/5997400/124044569-9153c080-d9c2-11eb-9639-1ec4e9d1c0f6.png)

## Workaround:

Right now, you just have to click the notification to see which DM or room the message came from.

## Platform:
Where is this issue occurring?

 Web ✔️
 iOS ✔️
 Android ✔️
 Desktop App ✔️
 Mobile Web

Basically anywhere that push/desktop notifications are enabled.

**Version Number:** Version 1.0.74-0
**Logs:** N/A
**Notes/Photos/Videos:** Screensots are included in `Expected Results` and `Actual Results` sections
**Expensify/Expensify Issue URL:** N/A

[View all open jobs on Upwork](https://www.upwork.com/ab/jobs/search/?q=Expensify%20React%20Native&sort=recency&user_location_match=2)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @conorpendergrast (`AutoAssignerTriage`), see https://stackoverflow.com/c/expensify/questions/4749 for more details.

**Comment by MelvinBot** [comment]
Triggered auto assignment to @roryabraham (`Engineering`), see https://stackoverflow.com/c/expensify/questions/4319 for more details.

**Comment by conorpendergrast** [comment]
Nice write-up Sasha; agreed that this is a confusing at the moment!

**Comment by trjExpensify** [comment]
Nice, love this. Linked it as a clean-up issue to the tracker. 👍

**Comment by roryabraham** [comment]
This will need to be internal, but I will hold onto it for now. I think I'll be able to get this done next week.

**Comment by roryabraham** [comment]
Still _hopefully_ will do this week, but I think this warrants `Weekly` priority instead of `Daily`.

**Comment by roryabraham** [comment]
Sorry I wasn't able to get this done. I imagine it should be pretty quick and easy, but I'm going to send it back to the pool while I'm OOO until July 26th. @sakluger Sorry for letting this slip through the cracks, but feel free to ping me if it's still not done by the time I get back and I can take care of this.

**Comment by nkuoch** [comment]
The QA failed on staging for the corresponding [PR](https://github.com/Expensify/Web-Expensify/pull/31585). So this issue probably won't be fixed after today's deploy. @HorusGoul will send a follow up PR.

**Comment by HorusGoul** [comment]
From this comment: https://github.com/Expensify/Web-Expensify/pull/31585#issuecomment-891730129

Weirdly, these chats don't have the required NVPs, and they should. In the photo, it appears as an announce room, but it doesn't have the `policyAnnounce` NVP.

Was this in a recently created room?

**Comment by jboniface** [comment]
@HorusGoul what's the latest on this? is there a particular engineer who can unblock it for you if it's still stuck?

**Comment by HorusGoul** [comment]
I can't reproduce the issue found in QA locally, so it's blocked there. The room where we tested this doesn't have the required NVPs to qualify as a room, even though it should.

**Comment by jboniface** [comment]
@roryabraham or @yuwenmemon can you assist with moving this along, please? I am still not 100% sure where we're stuck

@HorusGoul said:

> If new rooms don't have the chatType NVP, I guess it's a bug, we should open a separate issue to fix it. It could also be intentional because the feature it's in development, if someone could confirm, that would be great.

Do we need to spin out a new issue or what? It would be great if we could close this project off, and this is the last issue blocking us.

**Comment by yuwenmemon** [comment]
@HorusGoul Have you been able to reproduce the bug where new rooms don't have a `chatType` NVP?

**Comment by yuwenmemon** [comment]
@HorusGoul I double-checked the reportID mentioned in the [comment you linked](https://github.com/Expensify/Web-Expensify/pull/31585#issuecomment-891730129). I think there was a mistake grabbing which reportID because the comment mentions `78235338` but the actual reportID for the `#announce` room is `78235373`:
```JS
p = Policy.getCurrent();
Policy {id: "E741D686C2472FA1", owner: "applausetester+0802mna@applause.expensifail.com", permissions: "read,write,share,own", policy: {…}, _permissionChecksCache: {…}, …}
p.policy.chatReportIDAnnounce
78235373
```

As for your question here:
> The rooms I've created always had the NVPs. How are rooms created in staging/prod? I remember having to edit code to enable them for my local workspaces.

We will create #announce/#admins rooms for anyone that's on an expensify domain, `applause.expensifail.com` falls into this bucket. And, as evidenced by the fact that the policy has a `chatReportIDAnnounce` property, we did indeed create an announce room for the QA account. Perhaps it's possible that the QA tester sent a message in the wrong room?

I would say let's give this a spin yourself and see what happens?

**Comment by HorusGoul** [comment]
Done! I just confirmed that it works in production:

![photo_2021-09-06_14-06-34](https://user-images.githubusercontent.com/6759612/132215297-408036c4-ced5-41f6-bf6d-f7a362e43663.jpg)
